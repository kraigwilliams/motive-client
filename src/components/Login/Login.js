import React, { Component } from 'react'
import UserContext from '../../contexts/UserContext'
import {Required, FormLabel, FormInput, FormTitle, FormWrapper, LoginWrapper } from '../Form/Form'
import AuthApiService from '../../services/auth-api-service'
import {FormButton} from '../Button/Button'



export default class Login extends Component {
  static contextType = UserContext;
  static defaultProps = {
    location: {},
    history: {
      push: () => { },
    },
  }

  state ={ error: null }

  firstInput = React.createRef()


  handleLoginSuccess = () => {
    console.log('handle log in success firing');
    const { location, history } = this.props
    const destination = (location.state || {}).from || '/dashboard'
    history.push(destination)
  }

  hanldeSubmit = ev => {
    ev.preventDefault()
    const { username, password } = ev.target 

    this.setState({error: null})

    AuthApiService.postLogin({
      username: username.value,
      password: password.value,
    })
    .then(res => {
      username.value = ''
      password.value = ''
      this.context.processLogin(res.authToken)
      this.handleLoginSuccess()
    })
    .catch(res => {
      this.setState({ error: res.error })
    })
  }

  componentDidMount() {
    this.firstInput.current.focus()
  }

  render() {
    const { error } = this.state;
    return(
      <LoginWrapper>
        
        <FormWrapper onSubmit={this.hanldeSubmit}>
          <div>
            {error && <p>{error}</p>}
          </div>
          <FormTitle>
            Login
          </FormTitle>

          <FormLabel htmlFor='login-username-input'>
            Username 
          </FormLabel>
          <FormInput 
            placeholder='john_doe'
            ref={this.firstInput}
            id='login-username-input'
            name='username'
            aria-label="Enter your username"
            aria-required="true"
            required
          /><Required />

          <FormLabel htmlFor='login-password-input'>
            Password
          </FormLabel>
          <FormInput 
            type='password'
            placeholder='password'
            id='login-password-input'
            name='password'
            aria-label="Enter your password"
            aria-required="true"
            required
          /><Required />

          <FormButton type='submit'>
            Login
          </FormButton>
      </FormWrapper >
      
    </LoginWrapper>
    )
  }
}