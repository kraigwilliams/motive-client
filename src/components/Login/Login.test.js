import React from "react";
import ReactDOM from "react-dom";
import { mount, configure } from "enzyme";
import toJson from "enzyme-to-json";
import Login from "./Login";
import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });
describe(`<Login />`, () => {
  //Smoke Testing
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<Login />, div);
    ReactDOM.unmountComponentAtNode(div);
  });
  //Snapshot Testing
  it("renders Login by default", () => {
    const output = mount(<Login />);
    expect(toJson(output)).toMatchSnapshot();
  });
});
