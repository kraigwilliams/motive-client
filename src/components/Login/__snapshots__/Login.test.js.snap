// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> renders Login by default 1`] = `
<Login
  history={
    Object {
      "push": [Function],
    }
  }
  location={Object {}}
>
  <styled.div>
    <div
      className="sc-AxhCb faEozV"
    >
      <styled.form
        onChange={[Function]}
        onSubmit={[Function]}
      >
        <form
          className="sc-AxiKw lfvUnh"
          onChange={[Function]}
          onSubmit={[Function]}
        >
          <div
            style={
              Object {
                "height": "30px",
              }
            }
          >
            <Styled(Component)
              margin="10px 0px 20px 0px"
              onClick={[Function]}
              type="reset"
            >
              <ForwardRef
                className="sc-fzozJi jhvxvO"
                margin="10px 0px 20px 0px"
                onClick={[Function]}
                type="reset"
              >
                <button
                  className="Button sc-fzozJi jhvxvO"
                  margin="10px 0px 20px 0px"
                  onClick={[Function]}
                  type="reset"
                >
                  <FontAwesomeIcon
                    border={false}
                    className=""
                    fixedWidth={false}
                    flip={null}
                    icon="arrow-left"
                    inverse={false}
                    listItem={false}
                    mask={null}
                    pull={null}
                    pulse={false}
                    rotation={null}
                    size={null}
                    spin={false}
                    swapOpacity={false}
                    symbol={false}
                    title=""
                    transform={null}
                  >
                    <svg
                      aria-hidden="true"
                      className="svg-inline--fa fa-arrow-left fa-w-14 "
                      data-icon="arrow-left"
                      data-prefix="fas"
                      focusable="false"
                      role="img"
                      style={Object {}}
                      viewBox="0 0 448 512"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M257.5 445.1l-22.2 22.2c-9.4 9.4-24.6 9.4-33.9 0L7 273c-9.4-9.4-9.4-24.6 0-33.9L201.4 44.7c9.4-9.4 24.6-9.4 33.9 0l22.2 22.2c9.5 9.5 9.3 25-.4 34.3L136.6 216H424c13.3 0 24 10.7 24 24v32c0 13.3-10.7 24-24 24H136.6l120.5 114.8c9.8 9.3 10 24.8.4 34.3z"
                        fill="currentColor"
                        style={Object {}}
                      />
                    </svg>
                  </FontAwesomeIcon>
                </button>
              </ForwardRef>
            </Styled(Component)>
          </div>
          <styled.h1>
            <h1
              className="sc-AxirZ kOpkhk"
            >
              Login
            </h1>
          </styled.h1>
          <Styled(Label)
            htmlFor="login-username-input"
          >
            <Label
              className="sc-AxhUy ewJyDd"
              htmlFor="login-username-input"
            >
              <label
                className="Label sc-AxhUy ewJyDd"
                htmlFor="login-username-input"
              >
                Username 
                <Required>
                  <span
                    className="Required"
                  >
                    *
                  </span>
                </Required>
              </label>
            </Label>
          </Styled(Label)>
          <Styled(Component)
            aria-label="Enter your username"
            aria-required="true"
            id="login-username-input"
            name="username"
            placeholder="john_doe"
            required={true}
          >
            <ForwardRef
              aria-label="Enter your username"
              aria-required="true"
              className="sc-AxgMl ceOmaP"
              id="login-username-input"
              name="username"
              placeholder="john_doe"
              required={true}
            >
              <input
                aria-label="Enter your username"
                aria-required="true"
                className="Input sc-AxgMl ceOmaP"
                id="login-username-input"
                name="username"
                placeholder="john_doe"
                required={true}
                type="text"
              />
            </ForwardRef>
          </Styled(Component)>
          <Styled(Label)
            htmlFor="login-password-input"
          >
            <Label
              className="sc-AxhUy ewJyDd"
              htmlFor="login-password-input"
            >
              <label
                className="Label sc-AxhUy ewJyDd"
                htmlFor="login-password-input"
              >
                Password 
                <Required>
                  <span
                    className="Required"
                  >
                    *
                  </span>
                </Required>
              </label>
            </Label>
          </Styled(Label)>
          <Styled(Component)
            aria-label="Enter your password"
            aria-required="true"
            id="login-password-input"
            name="password"
            placeholder="password"
            required={true}
            type="password"
          >
            <ForwardRef
              aria-label="Enter your password"
              aria-required="true"
              className="sc-AxgMl ceOmaP"
              id="login-password-input"
              name="password"
              placeholder="password"
              required={true}
              type="password"
            >
              <input
                aria-label="Enter your password"
                aria-required="true"
                className="Input sc-AxgMl ceOmaP"
                id="login-password-input"
                name="password"
                placeholder="password"
                required={true}
                type="password"
              />
            </ForwardRef>
          </Styled(Component)>
          <Styled(Component)
            type="submit"
          >
            <ForwardRef
              className="sc-fzpans hrFzDy"
              type="submit"
            >
              <button
                className="Button sc-fzpans hrFzDy"
                type="submit"
              >
                Login
              </button>
            </ForwardRef>
          </Styled(Component)>
        </form>
      </styled.form>
    </div>
  </styled.div>
</Login>
`;
